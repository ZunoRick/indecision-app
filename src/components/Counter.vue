<template>
  <section>
    <h3>Counter: {{ counter }}</h3>
    <h3 data-testid="square-label">Square: {{ squareCounter }}</h3>

    <div>
      <button class="btn" @click="counter++">+1</button>
      <button class="btn" @click="counter--">-1</button>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { useCounter } from '@/composables/useCounter';
// import { computed, ref } from 'vue';

interface Props {
  value?: number;
}

const props = defineProps({
  value: { type: Number, required: true },
});

const { counter, squareCounter } = useCounter(props.value);

// const props = defineProps<Props>();

// const counter = ref(props.value ?? 2);

// const squareCounter = computed(() => counter.value * counter.value);
</script>

<style scoped>
@reference "../styles.css";
.btn {
  @apply p-5 mr-2 text-zinc-50 bg-blue-500 rounded hover:bg-blue-600;
}
</style>
