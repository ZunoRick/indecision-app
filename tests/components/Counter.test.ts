import { describe, expect, test } from 'vitest';
import { mount } from '@vue/test-utils';
import Counter from '@/components/Counter.vue';

describe('Pruebas en <Counter/>', () => {
  test('should match snapshot', () => {
    const wrapper = mount(Counter, {
      props: {
        value: 5,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('render the counter value correctly', () => {
    const value = 5;
    const wrapper = mount(Counter, {
      props: {
        value,
      },
    });
    const [counterLabel, squareLabel] = wrapper.findAll('h3');

    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${value * value}`,
    );

    expect(counterLabel.text()).toContain(`Counter: ${value}`);
    expect(squareLabel.text()).toContain(`Square: ${value * value}`);
  });
});
