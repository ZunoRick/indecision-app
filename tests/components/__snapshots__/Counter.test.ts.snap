// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en <Counter/> > should match snapshot 1`] = `
"<section data-v-9bd91872="">
  <h3 data-v-9bd91872="">Counter: 5</h3>
  <h3 data-v-9bd91872="" data-testid="square-label">Square: 25</h3>
  <div data-v-9bd91872=""><button data-v-9bd91872="" class="btn">+1</button><button data-v-9bd91872="" class="btn">-1</button></div>
</section>"
`;
